tests/test_interpreter.py

```diff
@@ -14,6 +22,10 @@
     register_impls,
 )
 from xdsl.interpreters.builtin import BuiltinFunctions
+from xdsl.interpreters.ptr import TypedPtr
+from xdsl.interpreters.shaped_array import ShapedArray
+from xdsl.interpreters.tensor import TensorFunctions
+from xdsl.interpreters.test import TestFunctions
```

Comment:
let's not import these, and only rely on functions defined in this test file

Reply:
For me the builtin dialect is an unfortunate exception just like in the rest of the framework :)

Reply:
okay, that means for op.operands != inputs i need to define another op for that right

Reply:
forget this, realised i was doing something dumb